gpytorch/lazy/block_diag_lazy_tensor.py

```diff
@@ -9,8 +10,26 @@
 from .block_lazy_tensor import BlockLazyTensor
 from .lazy_tensor import LazyTensor
 
+# metaclass of BlockDiagLazyTensor, overwrites behavior of constructor call
+# BlockDiagLazyTensor(base_lazy_tensor, block_dim=-3) to return a DiagLazyTensor
+# if base_lazy_tensor is a DiagLazyTensor itself
+class _MetaBlockDiagLazyTensor(ABCMeta):
+    def __call__(cls, base_lazy_tensor, block_dim=-3):
+        from .diag_lazy_tensor import DiagLazyTensor
+        if cls is BlockDiagLazyTensor and isinstance(base_lazy_tensor, DiagLazyTensor):
+            if not block_dim == -3:
+                raise NotImplementedError(
+                    f"Passing a base_lazy_tensor of type DiagLazyTensor to the constructor"
+                    "of BlockDiagLazyTensor with block_dim = {block_dim} != -3 is not supported."
```

Comment:
Got the f on the wrong line here. Also forgot a space :) 
```suggestion
                    "Passing a base_lazy_tensor of type DiagLazyTensor to the constructor "
                    f"of BlockDiagLazyTensor with block_dim = {block_dim} != -3 is not supported."
```